README.md

```diff
@@ -1,2 +1,325 @@
+
+This library includes the high level components providing complete payment forms to use the Paylike ecosystem.  
+If you are looking for our core elements required to implement payment flow towards the API, [check here](https://github.com/paylike/swift-engine).
+
+## Table of Contents
+---
+- [Paylike's Swift SDK](#paylikes-swift-sdk)
+  - [Table of Contents](#table-of-contents)
+  - [General Info](#general-info)
+    - [Purpose of this library](#purpose-of-this-library)
+    - [# Used Technologies, and iOS target](#used-technologies-and-ios-target)
+    - [Paylike API Reference](#paylike-api-reference)
+    - [Paylike Swift Ecosystem](#paylike-swift-ecosystem)
+  - [Usage](#usage)
+  - [TL;DR](#tldr)
+  - [Forms](#forms)
+    - [Simple Paymment Form](#simple-payment-form)
+      - [Using Custom Themes](#using-custom-themes)
+      - [Extending with additional data](#extending-with-additional-data)
+  - [Configurable Closures](#configurable-closures)
+    - [onSuccess](#onsuccess)
+    - [onError](#onerror)
+    - [beforePayment](#beforepayment)
+  - [Details](#details)
+    - [Localisation](#localisation)
+    - [Error Handling](#error-handling)
+  - [Example Application](#example-application)
+
+## General Info
+---
+### Purpose of this library
+This library was created to provide the simplest way of integrating Paylike into your iOS application.  
+It provides views and viewModels to create a payment form and an easy way to handle the payment result.
+
+### Used Technologies, and iOS target
+For the UI related design and implementation the chosen technology is **SwiftUI**.
+The minimal target is **iOS 13.0**. It is supported by 96.1% of iOS devices (Last Updated 2023.06.29) 
+
+### Paylike API Reference
+For the library you can find the API reference [here](https://paylike.io#todo-link).  
+To get more familiar with our server API you can find here the [official documentation](https://github.com/paylike/api-reference).
+
+### Paylike Swift Ecosystem
+
+<img title="Paylike Swift Ecosystem" alt="System architecture diagram of the Paylike Swift Ecosystem" src="readme-res/SysArch.png">
+
+## Usage
+---
+
+To use the ecosystem you must add the package as a dependency to your xcodeproject, or package.
+
+### Add To Project in XCode 14.3.1
+
+* Open your xcodeporj file
+* Go to the Package Dependencies tab
+* Click the + icon to add a new package
+* In the Search bar at the top right enter the package url: `https://github.com/paylike/swift-sdk`
+
+### Add to Package
+
+Package.swift
+```swift
+// swift-tools-version: 5.6
+
+import PackageDescription
+
+let package = Package(
+    // ...
+    dependencies: [
+        .package(url: "https://github.com/paylike/swift-sdk", .upToNextMajor(from: "0.1.0")),
+        // ...
+    ],
+    // ...
+)
+```
+
+## TL;DR
+---
+Don't forget to create your **OWN** API key (merchant ID) for the Paylike ecosystem. See: https://paylike.io/platforms/integration & https://github.com/paylike/api-docs#getting-an-api-key
```

Comment:
Should we add some custom text for the links? 
Like: "Integration Docs & Getting an API key"